### PR TITLE
fix network policies installed by orchestratord

### PR DESCRIPTION
**Pull request by doy-materialize**
### Motivation

we were previously testing in kind clusters which had a blanket policy of allowing all network traffic, and this caused a lot of the network policies we were writing to be incorrect. i fixed the setup for cloud's kind cluster in https://github.com/MaterializeInc/cloud/pull/10444 and with these changes i was able to successfully run several environments via orchestratord, both in the same namespace and in multiple different namespaces.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/orchestratord/src/controller/materialize/resources.rs [hunk 1]
```diff
@@ -485,6 +532,76 @@ fn create_network_policies(
...
+            spec: Some(NetworkPolicySpec {
+                egress: Some(vec![NetworkPolicyEgressRule {
+                    to: Some(vec![NetworkPolicyPeer {
+                        namespace_selector: Some(LabelSelector {
+                            match_labels: Some(btreemap! {
+                                "kubernetes.io/metadata.name".into()
+                                    => mz.namespace(),
+                            }),
+                            ..Default::default()
+                        }),
```
Comment (alex-hunt-materialize) [inline src/orchestratord/src/controller/materialize/resources.rs:584]: If we leave off the namespace selector, and just use a simplified pod selector of ```
{
    "materialize.cloud/app".to_owned() => mz.environmentd_service_name(),
}
```
could we instead have this just be installed once by the helm chart, allowing egress to environmentd in all namespaces?

It feels a bit gross creating resources across multiple namespaces, when the customer only created the Materialize CR in one. That is likely to break expectations, and it makes our code messier.

**Review by alex-hunt-materialize — APPROVED** [review]
This all works, but it would be nice to not do cross-namespace object creation when reconciling Materialize CRs.

File: src/catalog/tests/snapshots/debug__opened_trace.snap.new
Comment (doy-materialize) [inline src/catalog/tests/snapshots/debug__opened_trace.snap.new:1]: did you mean to include this file?

File: misc/helm-charts/operator/templates/networkpolicies.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{{- if .Values.networkPolicies.enabled }}
```
Comment (doy-materialize) [inline misc/helm-charts/operator/templates/networkpolicies.yaml:1]: this should also probably depend on `.Values.networkPolicies.internal.enabled`

File: src/orchestratord/src/controller/materialize.rs [hunk 3]
```diff
@@ -59,6 +59,8 @@ pub struct Args {
     #[clap(long)]
     enable_security_context: bool,
 
+    #[clap(long)]
+    orchestratord_pod_selector_labels: Vec<KeyValueArg<String, String>>,
```
Comment (doy-materialize) [inline src/orchestratord/src/controller/materialize.rs:63]: this can also go away since we're no longer using it

**Review by doy-materialize — COMMENTED** [review]
yeah, overall this definitely looks cleaner though, i didn't realize that normal NetworkPolicies let you do things across multiple namespaces like that

**Comment by alex-hunt-materialize** [comment]
> yeah, overall this definitely looks cleaner though, i didn't realize that normal NetworkPolicies let you do things across multiple namespaces like that

It took me a while to figure out that I needed an empty `namespaceSelector` to get it to go across namespaces. The docs for this are really terrible.

File: src/catalog/tests/snapshots/debug__opened_trace.snap.new
Comment (alex-hunt-materialize) [reply src/catalog/tests/snapshots/debug__opened_trace.snap.new:1]: nope

**Comment by doy-materialize** [comment]
all of the kubernetes documentation is surprisingly bad, given how much documentation there is

File: src/orchestratord/src/controller/materialize.rs (hunk 3 above)
Comment (alex-hunt-materialize) [reply src/orchestratord/src/controller/materialize.rs:63]: This is still used, but I did need to still pass it to orchestratord, which is now fixed.

**Pull request merged**